hydra-node/test/Hydra/Chain/Direct/Contract/Abort.hs

```diff
@@ -145,26 +145,28 @@ data AbortMutation
 genAbortMutation :: (Tx, UTxO) -> Gen SomeMutation
 genAbortMutation (tx, _utxo) =
   oneof
-    [ SomeMutation Nothing MutateParties . ChangeInputHeadDatum <$> do
+    [ SomeMutation (Just "burnt token number mismatch") MutateParties . ChangeInputHeadDatum <$> do
         moreParties <- (: healthyParties) <$> arbitrary
         c <- arbitrary
         pure $ Head.Initial c (partyToChain <$> moreParties) (toPlutusCurrencySymbol $ headPolicyId testSeedInput)
-    , SomeMutation Nothing DropOneCommitOutput
+    , SomeMutation (Just "reimbursed outputs dont match") DropOneCommitOutput
         . RemoveOutput
         <$> choose (0, fromIntegral (length (txOuts' tx) - 1))
-    , SomeMutation Nothing MutateThreadTokenQuantity <$> changeMintedValueQuantityFrom tx (-1)
-    , SomeMutation Nothing BurnOneTokenMore <$> addPTWithQuantity tx (-1)
-    , SomeMutation Nothing DropCollectedInput . RemoveInput <$> elements (txIns' tx)
+    , SomeMutation (Just "burnt token number mismatch") MutateThreadTokenQuantity <$> changeMintedValueQuantityFrom tx (-1)
+    , SomeMutation (Just "burnt token number mismatch") BurnOneTokenMore <$> addPTWithQuantity tx (-1)
+    , -- REVIEW: why here are we getting back a trace from HeadTokens?
```

Comment:
This is fine and expected. The minting (HeadTokens) is the one which actually checks the number of tokens being burned (right now). 

In the abort transactions up to 4 scripts run! Head, init, commit and the minting policy script.

Reply:
In fact, our head token script does check too much. Or things which are redundant. According to the spec we do not need to do these checks in the minting policy as the v_head is already doing it.

Let's address this as part of #720

Reply:
Outdated.

Addressed as part of #720.